File: 백준/Silver/1094. 막대기/막대기.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int x = toInt(br.readLine()), stick = 64;
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        heap.add(stick);
        int sum = 64;

        while (sum != x) {
            int halfStick = heap.poll() / 2;
            sum -= halfStick * 2;
            if (sum + halfStick >= x) {
                heap.add(halfStick);
                sum += halfStick;
            } else {
                heap.add(halfStick);
                heap.add(halfStick);
                sum += halfStick*2;
            }
        }

        System.out.println(heap.size());
    }

    public static int toInt(String s) {
        return Integer.parseInt(s);
    }
}

File: 백준/Silver/1094. 막대기/막대기.py
import heapq

x = int(input())

stick = 64
heap = []
heapq.heappush(heap, stick)

while sum(heap) != x:
    shortest_stick = heapq.heappop(heap)
    half_stick = shortest_stick//2
    if sum(heap) + half_stick >= x:
        heapq.heappush(heap, half_stick)
    else:
        heapq.heappush(heap, half_stick)
        heapq.heappush(heap, half_stick)
        
print(len(heap))